src/Altinn.App.Core/Internal/Validation/IValidationService.cs

```diff
@@ -41,6 +41,12 @@ public interface IValidationService
     /// <returns>List of validation issues for this data element</returns>
     Task<List<ValidationIssue>> ValidateDataElement(Instance instance, DataElement dataElement, DataType dataType, string? language);
 
+    /// <summary>
+    /// Validates a file upload. Will prevent the file from being stored if
+    /// issues with severity Error are found.
+    /// </summary>
+    Task<List<ValidationIssue>> ValidateFileUpload(Instance instance, DataType dataType, byte[] fileContent, string? filename, string? mimeType, string? language);
```

Comment:
Might be too exotic, but if we used `ReadOnlyMemory<byte>` here that might be a better API in terms of
* Communicating through the typesystem the expected use of `fileContent` here (the validator should only need to read)
* Would allow us to potentially pool buffers in the future (give us more flexibility for managing perf and allocations)

Reply:
Yeah this kind of thing works better if its' use is consistent throughout, so might be better to do a larger breaking change once. Atleast most (if not all) of BCL understands ROM/ROS by now